### ISSUE TITLE
Disconnected Communities in Parallel Leiden

### ISSUE THREAD

**Issue by CxVercility** [networkit/networkit#1244]
Hi there,
It's come to my attention that while benchmarking against my own implementation of the Leiden community algorithm, which was merged into networkit a while ago, someone has found that it's able to produce disconnected communities. (https://arxiv.org/html/2312.13936v5)

As this is not possible by design of the Leiden algorithm itself this is likely an implementation bug.
Im unable to confirm this but I'm a little embarassed to say that I might actually have forgotten to test for disconnected communities ..

If noone has time or the knowledge to confirm/test this you might want to remove the algorithm from the library (or provide a warning)

Greets

**Comment by fabratu** [comment]
Hi, sry for the late reply. I have looked into GVE-Leiden and based on their findings (both concerning the modularity scores as the erroneous results) we will remove the current Leiden implementation from the next release. The rationale here is, that it will indeed take some time to fix the problems. The plan is though to have a revised implementation in near future.

**Comment by fabratu** [comment]
Update: Instead of removing the algorithm, we opted for a warning + documentation until the error is fixed. I will close this issue.

**Comment by adsharma** [comment]
@fabratu Please see the discussion here:

https://github.com/nodef/leiden-communities-openmp.py/pull/1

What are the problems in incorporating the GVE Leiden code itself into NetworkIt? Both are C++ and OpenMP based and license compatible.
